### PR TITLE
fix: Solves Teradata `SSLMODE` support and adds support for additional `teradatasql` connection options as a JSON string

### PR DESCRIPTION
**Pull request by jcqpinto**
Solves Teradata SSLMODE support - connection timeout issue (#1011) and adds support for additional teradatasql connection options as a JSON string.

Example of usage:
`data-validation connections add -c my_td Teradata --json-params "{'sslmode': 'disable', 'encryptdata': 'true'}"`

### CODE REVIEW

**Review by helensilva14 — COMMENTED** [review]
LGTM! Pending review from @nehanene15 and maybe @Raniksingh as well.

**Review by nehanene15 — APPROVED** [review]
LGTM!

**Review by Raniksingh — APPROVED** [review]

**Pull request merged**
